pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -119,107 +117,139 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
+	// Add conditions if there are none
+	if instance.Status.Conditions == nil {
+		initConditions(&instance.Status.Conditions)
+
+		err = r.client.Status().Update(context.TODO(), instance)
+		if err != nil {
+			reqLogger.Error(err, "Failed to add conditions to status")
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Create happy conditions
+	initHappyConditions(&r.conditions)
+
 	for _, f := range []func(*hcov1alpha1.HyperConverged, logr.Logger, reconcile.Request) error{
 		r.ensureKubeVirtConfig,
 		r.ensureKubeVirt,
 		r.ensureCDI,
 		r.ensureNetworkAddons,
-		r.ensureKubeVirtCommonTemplatebundle,
+		r.ensureKubeVirtCommonTemplateBundle,
 		r.ensureKubeVirtNodeLabellerBundle,
 		r.ensureKubeVirtTemplateValidator,
 	} {
 		err = f(instance, reqLogger, request)
 		if err != nil {
+			initReconcileFailureConditions(&instance.Status.Conditions)
+			// don't want to overwrite the actual reconcile failure
+			uErr := r.client.Status().Update(context.TODO(), instance)
```

Comment:
For a first pass context.TODO will be ok, but we should quickly turn around to fix.  Reconcile can be executed much faster than the API server can process the requests and we don't want to have any races with our state.